src/main.rs

```diff
@@ -240,6 +271,129 @@ async fn main() -> Result<()> {
                 .collect::<Vec<_>>();
             println!("Groups found: {:?}", group_ids)
         }
+        CommandLineArgs::FileEncrypt {
+            filename: infile,
+            device_path,
+            users: user_ids,
+            groups: group_ids,
+            output: maybe_output,
+        } => {
+            let file = std::fs::read(infile.clone())?;
+            println!("Read in file {:?}", infile);
+            let sdk = initialize_sdk_from_file(&device_path).await?;
+            let output = match maybe_output {
+                // User specified an output path.
+                Some(desired) => {
+                    // User specified a directory for output with no filename
+                    if desired.is_dir() {
+                        // This is the file component of infile.
+                        let mut filename = match infile.file_name() {
```

Comment:
This chunk also seems easily refactored out. `maybe_filename_to_encrypted_name` or something.

Reply:
I cleaned this up with `ok_or()` instead